src/Datasource/ModelAwareTrait.php

```diff
@@ -35,7 +35,10 @@ trait ModelAwareTrait
      * Plugin classes should use `Plugin.Comments` style names to correctly load
      * models from the correct plugin.
      *
-     * @var string
+     * Use false to not use auto-loading on this object. Null auto-detects based on
+     * controller name.
+     *
+     * @var string|false|null
```

Comment:
I have the feeling the static analyses will be unhappy with this in 4.x.

Reply:
We can use empty string and strict === '' check here for fasely value.

Reply:
I can fix that up after merge for 4.x as target.